### PR TITLE
Resources migrated and excluded by MTC

### PR DESCRIPTION
**Pull request by RichardHoch**
Resolves: https://issues.redhat.com/browse/MIG-417
Content based on the upstream content approved in https://github.com/konveyor/mig-operator/pull/561/files and comments on the Jira. 

Previews: https://deploy-preview-29096--osdocs.netlify.app/openshift-enterprise/latest/migration/migrating_3_4/migrating-application-workloads-3-4.html#about-resources-mtc-migrates_migrating-3-4  
https://deploy-preview-29096--osdocs.netlify.app/openshift-enterprise/latest/migration/migrating_3_4/migrating-applications-with-cam-3-4.html#migration-excluding-resources_migrating-3-4

The same content is found in: https://deploy-preview-29096--osdocs.netlify.app/openshift-enterprise/latest/migration/migrating_4_1_4/migrating-application-workloads-4-1-4.html#about-resources-mtc-migrates_migrating-4-1-4
https://deploy-preview-29096--osdocs.netlify.app/openshift-enterprise/latest/migration/migrating_4_1_4/migrating-applications-with-cam-4-1-4.html#migration-excluding-resources_migrating-4-1-4

and

https://deploy-preview-29096--osdocs.netlify.app/openshift-enterprise/latest/migration/migrating_4_2_4/migrating-application-workloads-4-2-4.html#about-resources-mtc-migrates_migrating-4-2-4
https://deploy-preview-29096--osdocs.netlify.app/openshift-enterprise/latest/migration/migrating_4_2_4/migrating-applications-with-cam-4-2-4.html#migration-excluding-resources_migrating-4-2-4

### CODE REVIEW

**Comment by RichardHoch** [comment]
@apinnick : please review.  I debated a bit about where to place the discussion of the resources migrated by the MTC. It might also be included in the later section "Migrating your applications" (https://deploy-preview-29096--osdocs.netlify.app/openshift-enterprise/latest/migration/migrating_4_2_4/migrating-applications-with-cam-4-2-4.html and the parallel section in the other MTC documents). Let me know which location you prefer or suggest another.

File: modules/migration-understanding-resources-mtc-migrates.adoc [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// Module included in the following assemblies:
+//
+// * migration/migrating_3_4/migrating-application-workloads-3-4.adoc
+// * migration/migrating_4_1_4/migrating-application-workloads-4-1-4.adoc
+// * migration/migrating_4_2_4/migrating-application-workloads-4-2-4.adoc
+
+[id='about-resources-mtc-migrates_{context}']
```
Comment (apinnick) [inline modules/migration-understanding-resources-mtc-migrates.adoc:7]: ID is supposed to be the same as file name. Use "about" instead of "understanding".

File: modules/migration-understanding-resources-mtc-migrates.adoc [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+= About the resources migrated
+[role="_abstract"]
+
+You can use the {mtc-full} ({mtc-short}) to migrate the following types of resources:
+
+* Namespaces specified in migration plans. Migration plans have the granularity of a namespece.
```
Comment (apinnick) [inline modules/migration-understanding-resources-mtc-migrates.adoc:13]: typo in last word

File: modules/migration-understanding-resources-mtc-migrates.adoc [hunk 3]
```diff
@@ -0,0 +1,19 @@
+// Module included in the following assemblies:
+//
+// * migration/migrating_3_4/migrating-application-workloads-3-4.adoc
+// * migration/migrating_4_1_4/migrating-application-workloads-4-1-4.adoc
+// * migration/migrating_4_2_4/migrating-application-workloads-4-2-4.adoc
+
+[id='about-resources-mtc-migrates_{context}']
+= About the resources migrated
```
Comment (apinnick) [inline modules/migration-understanding-resources-mtc-migrates.adoc:8]: Title is extremely vague. Needs to be more specific.

File: modules/migration-understanding-resources-mtc-migrates.adoc [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+[role="_abstract"]
+
+You can use the {mtc-full} ({mtc-short}) to migrate the following types of resources:
+
+* Namespaces specified in migration plans. Migration plans have the granularity of a namespece.
+
+* Namespace-scoped resources. When a namespace is selected for migration, then all the objects and resources (services, pods, etc.) pertaining to that namespace are also selected for migration. Additionally, if a resource scoped by a namespace depends on a resource that is scoped by a cluster, then the resource scoped by the cluster is also migrated by the  {mtc-full} Operator.
```
Comment (apinnick) [inline modules/migration-understanding-resources-mtc-migrates.adoc:15]: Rewrite to remove fluff. Avoid using parentheses. They can be problematic for localization.

Remove "then". It does not add anything to the meaning, AFAIK.

The resources are not migrated by the operator. The SME is using 'operator' to refer to the VMIO component. It's just MTC to the user.

File: modules/migration-understanding-resources-mtc-migrates.adoc [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
++
+For instance, a Security Context Constraint (SCC) is a resource scoped by a cluster and a service account (SA) is a resource scoped by a namespece. If an SA exists in a namespace that you select for migration, then the {mtc-full} Operator automatically locates any SCCs that are applied to this SA and include those SCCs in the migration as well. Similarly, the {mtc-full} Operator migrates Persistent Volume Claims linked to the Persistent Volumes of the namespace.
```
Comment (apinnick) [inline modules/migration-understanding-resources-mtc-migrates.adoc:17]: Change "For instance" to something less colloquial -- "For example".
Remove ", then ..." here and elsewhere.

Lower case full names of SCC, PVs, PVCs. See the OpenShift docs glossary.

Operator does not do anything here. Operator only installs and provisions MTC. This is not the MTC Operator.

File: modules/migration-understanding-resources-mtc-migrates.adoc [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+* Custom Resources (CRs) and Custom Resource Definitions (CRDs). Any CRs that are scoped by a namespace are automatically included in the migration of the namespace. Therefore, the {mtc-full} Operator also migrates the CRDs associated with those CRs.
```
Comment (apinnick) [inline modules/migration-understanding-resources-mtc-migrates.adoc:19]: Remove "Any" and "Therefore". There is no cause-and-effect relationship. Remove "also".
No Operator. MTC performs the migration.

File: modules/migration-excluding-resources.adoc [hunk 7]
```diff
@@ -6,7 +6,9 @@
 [id='migration-excluding-resources_{context}']
 = Excluding resources from a migration plan
 
-You can exclude resources, for example, image streams, persistent volumes (PVs), or subscriptions, from a  {mtc-full} ({mtc-short}) migration plan in order to reduce the load or to migrate images or PVs with a different tool.
+You can exclude resources, for example, image streams, persistent volumes (PVs), or subscriptions, from a {mtc-full} ({mtc-short}) migration plan in order to reduce the load or to migrate images or PVs with a different tool.
+
+Service catalog resources and Operator Lifecycle Manager (OLM) resources are excluded by default from migration by the {mtc-full} operator. OLM migration itself is not handled by the {mtc-full} operator.
```
Comment (apinnick) [inline modules/migration-excluding-resources.adoc:11]: Not the operator. Operator does not perform migration. MTC does.

File: modules/migration-excluding-resources.adoc [hunk 8]
```diff
@@ -41,10 +43,12 @@ spec:
   - serviceclasses
   - serviceinstances
   - serviceplans
+  - operatorgroups
+  - events
 ----
 <1> Add `disable_image_migration: true` to exclude image streams from the migration. Do not edit the `excluded_resources` parameter. `imagestreams` is added to `excluded_resources` when the `MigrationController` pod restarts.
 <2> Add `disable_pv_migration: true` to exclude PVs from the migration plan. Do not edit the `excluded_resources` parameter. `persistentvolumes` and `persistentvolumeclaims` are added to `excluded_resources` when the `MigrationController` pod restarts. Disabling PV migration also disables PV discovery when you create the migration plan.
-<3> You can add {product-title} resources to the `excluded_resources` list. Do not delete any of the default excluded resources. These resources are known to be problematic for migration.
+<3> You can add {product-title} resources to the `excluded_resources` list. Default excluded resources are shown. These resources are known to be problematic for migration. Do not delete any of them.
```
Comment (apinnick) [inline modules/migration-excluding-resources.adoc:51]: Why did you change active to passive voice in second sentence? What was the problem with the original?

File: modules/migration-understanding-resources-mtc-migrates.adoc [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+//
+// * migration/migrating_3_4/migrating-application-workloads-3-4.adoc
+// * migration/migrating_4_1_4/migrating-application-workloads-4-1-4.adoc
+// * migration/migrating_4_2_4/migrating-application-workloads-4-2-4.adoc
+
+[id='about-resources-mtc-migrates_{context}']
+= About the resources migrated
+[role="_abstract"]
+
+You can use the {mtc-full} ({mtc-short}) to migrate the following types of resources:
```
Comment (apinnick) [inline modules/migration-understanding-resources-mtc-migrates.adoc:11]: No, we are not telling the user what they can do. We are telling them the limitations of MTC. This sentence should be declarative: MTC migrates the following resources.

**Comment by apinnick** [comment]
> @apinnick : please review. I debated a bit about where to place the discussion of the resources migrated by the MTC. It might also be included in the later section "Migrating your applications" (https://deploy-preview-29096--osdocs.netlify.app/openshift-enterprise/latest/migration/migrating_4_2_4/migrating-applications-with-cam-4-2-4.html and the parallel section in the other MTC documents). Let me know which location you prefer or suggest another.

Location is OK since this is conceptual material.

File: modules/migration-excluding-resources.adoc (hunk 8 above)
Comment (RichardHoch) [reply modules/migration-excluding-resources.adoc:51]: I moved the second sentence; I didn't change its voice. I added a new sentence to make it clear that the resources shown were the default resources, just in case the user might think they were samples. If you think that's obvious from the original, I'll get rid of the changes.

Comment (apinnick) [reply modules/migration-excluding-resources.adoc:51]: Who wrote "are shown"? That is what I think should be rewritten.

Comment (apinnick) [reply modules/migration-excluding-resources.adoc:51]: I think this is clearer:

You can add {product-title} resources to the `excluded_resources` list. Do not delete the default excluded resources. These resources are problematic to migrate and must be excluded.

**Comment by RichardHoch** [comment]
@apinnick : am still waiting for information about "OLM migration" from the SME but I made the other changes you recommended. Perhaps we can delete the sentence about "OLM migration." I'm not sure what it actually adds.

**Comment by RichardHoch** [comment]
@apinnick : please review. Incorporates SME response to question about "OLM migration" and rebasing.

File: modules/migration-excluding-resources.adoc [hunk 10]
```diff
@@ -6,7 +6,9 @@
 [id='migration-excluding-resources_{context}']
 = Excluding resources from a migration plan
 
-You can exclude resources, for example, image streams, persistent volumes (PVs), or subscriptions, from a  {mtc-full} ({mtc-short}) migration plan in order to reduce the load or to migrate images or PVs with a different tool.
+You can exclude resources, for example, image streams, persistent volumes (PVs), or subscriptions, from a {mtc-full} ({mtc-short}) migration plan in order to reduce the load or to migrate images or PVs with a different tool.
```
Comment (kalexand-rh) [inline modules/migration-excluding-resources.adoc:9]:
```suggestion
You can exclude resources, for example, image streams, persistent volumes (PVs), or subscriptions, from a {mtc-full} ({mtc-short}) migration plan to reduce the resource load for migration or to migrate images or PVs with a different tool.
```

File: modules/migration-mtc-workflow.adoc [hunk 11]
```diff
@@ -7,7 +7,19 @@
 [id='migration-mtc-workflow_{context}']
 = {mtc-full} workflow
 
-Migrating an application with the {mtc-short} web console involves the following workflow:
+The {mtc-full} ({mtc-short}) enables you to migrate Kubernetes resources, persistent volume data, and internal container images from an {product-title} source cluster to an {product-title} {product-version} target cluster, using the {mtc-short} web console or the Kubernetes API.
```
Comment (kalexand-rh) [inline modules/migration-mtc-workflow.adoc:10]:
```suggestion
You use the {mtc-full} ({mtc-short}) to migrate Kubernetes resources, persistent volume data, and internal container images from an {product-title} source cluster to an {product-title} {product-version} target cluster by using the {mtc-short} web console or the Kubernetes API.
```

File: modules/migration-mtc-workflow.adoc [hunk 12]
```diff
@@ -7,7 +7,19 @@
...
+
+The ({mtc-short}) migrates the following resources:
+
+* A namespace specified in a migration plan.
+
+* Namespace-scoped resources: When the {mtc-short} migrates a namespace, it migrates all the objects and resources associated with that namespace, such as services or pods. Additionally, if a resource that exists in the namespace but not at the cluster level depends on a resource that exists at the cluster level, the {mtc-short} migrates both resources.
++
+For example, a security context constraint (SCC) is a resource that exists at the cluster level and a service account (SA) is a resource that exists at the namespace level. If an SA exists in a namespace that the {mtc-short} migrates, the {mtc-short} automatically locates any SCCs that are linked to the SA and also migrates those SCCs. Similarly, the {mtc-short} migrates persistent volume claims linked to the persistent volumes of the namespace.
```
Comment (kalexand-rh) [inline modules/migration-mtc-workflow.adoc:18]:
```suggestion
For example, a security context constraint (SCC) is a resource that exists at the cluster level and a service account (SA) is a resource that exists at the namespace level. If an SA exists in a namespace that the {mtc-short} migrates, the {mtc-short} automatically locates any SCCs that are linked to the SA and also migrates those SCCs. Similarly, the {mtc-short} migrates persistent volume claims that are linked to the persistent volumes of the namespace.
```

File: modules/migration-mtc-workflow.adoc [hunk 13]
```diff
@@ -7,7 +7,19 @@
...
+
+* Custom resources (CRs) and custom resource definitions (CRDs): The {mtc-short} automatically migrates any CRs that exist at the namespace level as well as the CRDs linked to those CRs.
```
Comment (kalexand-rh) [inline modules/migration-mtc-workflow.adoc:20]:
```suggestion
* Custom resources (CRs) and custom resource definitions (CRDs): The {mtc-short} automatically migrates any CRs that exist at the namespace level as well as the CRDs that are linked to those CRs.
```

File: modules/migration-prerequisites.adoc [hunk 14]
```diff
@@ -16,34 +16,5 @@ endif::[]
...
-* `dotnet: 2.2`
-* `dotnet-runtime: 2.2`
-* `nginx: 1.12`
-* `nodejs: 8, 8-RHOAR, 10-SCL`
-* `perl:5.24`
-* `php: 7.0, 7.1`
-* `redis: 3.2`
-endif::[]
+* The cluster on which the `MigrationController` CR is installed must have unrestricted network access to the other clusters.
+* If your application uses images from the `openshift` namespace, the required versions of the images must be present on the target cluster. You can manually update an image stream tag in order to use a deprecated {product-title} 3 image on an {product-title} {product-version} cluster.
```
Comment (kalexand-rh) [inline modules/migration-prerequisites.adoc:20]:
```suggestion
* If your application uses images from the `openshift` namespace, the required versions of the images must be present on the target cluster. You can manually update an image stream tag to use a deprecated {product-title} 3 image on an {product-title} {product-version} cluster.
```

File: modules/migration-updating-internal-images.adoc [hunk 15]
```diff
@@ -0,0 +1,67 @@
+// Module included in the following assemblies:
+//
+// * migration/migrating_3_4/migrating-applications-with-cam-3-4.adoc
+
+[id='migration-updating-internal-images_{context}']
+= Updating internal images manually
+
+If your application uses images from the `openshift` namespace, the required versions of the images must be present on the target cluster.
+
+If the {product-title} 3 image is deprecated in {product-title} {product-version}, you can manually update the image stream tag using Podman.
```
Comment (kalexand-rh) [inline modules/migration-updating-internal-images.adoc:10]:
```suggestion
If the {product-title} 3 image is deprecated in {product-title} {product-version}, you can manually update the image stream tag by using Podman.
```

File: modules/migration-updating-internal-images.adoc [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+
+.Prerequisites
+
+* You must have Podman installed.
+* You must have `cluster-admin` privileges.
+
+.Procedure
+
+. Expose the internal registries on the source and target clusters.
```
Comment (kalexand-rh) [inline modules/migration-updating-internal-images.adoc:19]: Is this enough information for a user to expose the registries?

File: modules/migration-updating-internal-images.adoc [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+. If you are using insecure registries, add your registry host values to the `[registries.insecure]` section of `/etc/container/registries.conf` to ensure that Podman does not encounter a TLS verification error.
```
Comment (kalexand-rh) [inline modules/migration-updating-internal-images.adoc:20]: I think this step needs an example.
On which host is the `/etc/container/registries.conf` file?

File: modules/migration-updating-internal-images.adoc [hunk 18]
```diff
@@ -0,0 +1,67 @@
...
+= Updating internal images manually
+
+If your application uses images from the `openshift` namespace, the required versions of the images must be present on the target cluster.
+
+If the {product-title} 3 image is deprecated in {product-title} {product-version}, you can manually update the image stream tag using Podman.
+
+.Prerequisites
+
+* You must have Podman installed.
+* You must have `cluster-admin` privileges.
```
Comment (kalexand-rh) [inline modules/migration-updating-internal-images.adoc:15]: On which cluster?

File: modules/migration-updating-internal-images.adoc [hunk 19]
```diff
@@ -0,0 +1,67 @@
...
+[id='migration-updating-internal-images_{context}']
+= Updating internal images manually
+
+If your application uses images from the `openshift` namespace, the required versions of the images must be present on the target cluster.
+
+If the {product-title} 3 image is deprecated in {product-title} {product-version}, you can manually update the image stream tag using Podman.
+
+.Prerequisites
+
+* You must have Podman installed.
```
Comment (kalexand-rh) [inline modules/migration-updating-internal-images.adoc:14]:
```suggestion
* You must have Podman installed on your local host.
```

File: modules/migration-updating-internal-images.adoc [hunk 20]
```diff
@@ -0,0 +1,67 @@
...
+
+. Expose the internal registries on the source and target clusters.
+. If you are using insecure registries, add your registry host values to the `[registries.insecure]` section of `/etc/container/registries.conf` to ensure that Podman does not encounter a TLS verification error.
+. Log in to the {product-title} 3 registry:
++
+[source,terminal]
+----
+$ podman login -u $(oc whoami) -p $(oc whoami -t) --tls-verify=false <ocp3_host>
+----
+
```
Comment (kalexand-rh) [inline modules/migration-updating-internal-images.adoc:27]:
```suggestion
<1> `<source_registry_host>` is the _________ for the {product-title} 3 registry.

```
Is this a fully qualified domain name or IP address or some other thing?

File: modules/migration-updating-internal-images.adoc [hunk 21]
```diff
@@ -0,0 +1,67 @@
...
+. Log in to the {product-title} 4 registry:
++
+[source,terminal]
+----
+$ podman login -u $(oc whoami) -p $(oc whoami -t) --tls-verify=false <ocp4_host>
```
Comment (kalexand-rh) [inline modules/migration-updating-internal-images.adoc:32]: Update this command to be similar to the previous command.

File: modules/migration-updating-internal-images.adoc [hunk 22]
```diff
@@ -0,0 +1,67 @@
...
+----
+
+. Pull the deprecated image, for example, `python:3.3`:
```
Comment (kalexand-rh) [inline modules/migration-updating-internal-images.adoc:35]: I strongly suggest not using a specific package in the only version of the command that you present. It's more important that you're clear that the user needs to specify both the image name and version. You can add an example that uses `python:3.3`, if you want.

File: modules/migration-updating-internal-images.adoc [hunk 23]
```diff
@@ -0,0 +1,67 @@
...
+
+.Procedure
+
+. Expose the internal registries on the source and target clusters.
+. If you are using insecure registries, add your registry host values to the `[registries.insecure]` section of `/etc/container/registries.conf` to ensure that Podman does not encounter a TLS verification error.
+. Log in to the {product-title} 3 registry:
++
+[source,terminal]
+----
+$ podman login -u $(oc whoami) -p $(oc whoami -t) --tls-verify=false <ocp3_host>
```
Comment (kalexand-rh) [inline modules/migration-updating-internal-images.adoc:25]: "OCP" is not an approved shortname, and the only time it's acceptable to include it in the collection is in output or strings that we can't change.

You need to pull it from all you variables.
```suggestion
$ podman login -u $(oc whoami) -p $(oc whoami -t) --tls-verify=false <source_registry_host> <1>
```

File: modules/migration-updating-internal-images.adoc [hunk 24]
```diff
@@ -0,0 +1,67 @@
...
+[source,terminal]
+----
+$ podman login -u $(oc whoami) -p $(oc whoami -t) --tls-verify=false <ocp4_host>
+----
+
+. Pull the deprecated image, for example, `python:3.3`:
++
+[source,terminal]
+----
+$ podman pull <ocp3_host>/openshift/python:3.3
```
Comment (kalexand-rh) [inline modules/migration-updating-internal-images.adoc:39]: In this, and all subsequent steps, you need to explain what `<ocp3/4_host>` is. And not use "ocp."

File: modules/migration-updating-internal-images.adoc [hunk 25]
```diff
@@ -0,0 +1,67 @@
...
+[source,terminal]
+----
+$ podman push <ocp4_host>/openshift/python:3.3
+----
+
+. Verify that the image has a valid image stream on the {product-title} 4 cluster:
++
+[source,terminal]
+----
+$ oc get imagestream -n openshift | grep python
```
Comment (kalexand-rh) [inline modules/migration-updating-internal-images.adoc:60]:
```suggestion
$ oc get imagestream -n openshift | grep <image_name>
```

**Review by kalexand-rh — COMMENTED** [review]
I have some edits and suggestions.

File: modules/migration-mtc-workflow.adoc (hunk 11 above)
Comment (kalexand-rh) [reply modules/migration-mtc-workflow.adoc:10]: @apinnick, what do you think of the suggested edit?

File: modules/migration-updating-internal-images.adoc (hunk 24 above)
Comment (apinnick) [reply modules/migration-updating-internal-images.adoc]: rhocp_host

File: modules/migration-mtc-workflow.adoc (hunk 11 above)
Comment (apinnick) [reply modules/migration-mtc-workflow.adoc:10]: Fine
